test/python/transpiler/test_optimize_1q_gates.py

```diff
@@ -287,5 +287,54 @@ def test_optimize_u3_to_u1_round(self):
         self.assertEqual(expected, result)
 
 
+class TestOptimize1qGatesBasis(QiskitTestCase):
+    """Test for 1q gate optimizations parameter reduction with basis """
+
```

Comment:
Test cases for u1 reduction and e.g. `basis=['u2', 'u3']`?

Reply:
Sure.

`u1` will be reduced to `u3` in this case. if `u3` is not part of basis, exception. I added two tests with those situations in e57860d33.